test-data/unit/check-literal.test

```diff
@@ -1797,7 +1797,8 @@ def f5(x: Literal[1], y: Union[Literal[1], Literal[2]]) -> None: pass
 def f6(x: Optional[Literal[1]], y: Optional[Literal[2]]) -> None: pass
 
 reveal_type(unify(f1))  # N: Revealed type is "Literal[1]"
-reveal_type(unify(f2))  # N: Revealed type is "<nothing>"
+if object():
```

Comment:
This seems like a regression to me, because this is not "really `<nothing>`, execution is stopped", it is just "we cannot do a better job with two literal values".

In the long run we need to figure out how to tell these two cases apart.

Reply:
This was always the case for function scope. I only made module scope work the same way.

I agree that entire behavior leaves a lot to be desired, and I'll try to work on that next.

It's also broken that "cannot call function of unknown type" does not have that same treatment of "oh no, we've seen an uninhibited type!".